Source/player.cpp

```diff
@@ -787,10 +786,9 @@ bool DamageWeapon(int pnum, int durrnd)
 			return false;
 		}
 
-		player.InvBody[INVLOC_HAND_RIGHT]._iDurability--;
+		if (player.InvBody[INVLOC_HAND_RIGHT]._iDurability != 0)
+			player.InvBody[INVLOC_HAND_RIGHT]._iDurability--;
 		if (player.InvBody[INVLOC_HAND_RIGHT]._iDurability == 0) {
-			NetSendCmdDelItem(true, INVLOC_HAND_RIGHT);
-			player.InvBody[INVLOC_HAND_RIGHT]._itype = ItemType::None;
 			CalcPlrInv(player, true);

```

Comment:
If you wanted to make this an optional feature you could restore the `DelItem` call (and setting the relevant item to None) but gate it behind an if check. Something like:
```cpp
if (durability == 0) {
    if (!sgOptions.allowZeroDurabilityItems) {
        NetSendCmdDelItem(...);
        player.InvBody[<location>]._itype = ItemType::None;
    }
    CalcPlrInv(...);
    return true
}